wordpress-mobile/WordPress-iOS
# Add option to customize the app accent color

Fixes: #19966 

## Description
This PR adds option to customize app wide "Accent Color" within "App Settings".

UI changes include:
- introduce "Appearance" section on the "App Settings" screen 
- add new "Accent Color" setting which navigates to a color list for selecting user preferred accent color
- existing semantic colors: `primary`, `accent`, `brand` will now use `AppColor.accentColor` under the hood
- several existing screens / views are now listening to the `appColorDidUpdateAccent` notification in order to properly update accent color when it is changed

Majority of the new logic is defined in new `AppColor` type and it is static, since we want to access this from the existing static variables related to colors which are already used across the app.

| App Settings | Accent Color |
| - | - |
| <img width=320 src="https://user-images.githubusercontent.com/2762374/226314999-77ae4c66-2f68-469a-9e3a-6ace9c252fd9.png" /> | <img width=320 src="https://user-images.githubusercontent.com/2762374/226315013-0f1fb85b-61d0-4c5b-b71a-e9afc98a5910.png" /> | 

## Testing

- Install Wordpress or Jetpack and login
- Navigate to App Settings / Accent Color
- Choose custom Accent Color from the predefined list
- Verify that custom Accent Color is properly used across the app

## Regression Notes
1. Potential unintended areas of impact
    - it is possible that we'll encounter one-off bugs (ie. unexpected tint color on some screen) since semantic colors in code are not perfectly consolidated (this also depends on the quality of existing implementation), but I'd suggest to tackle these minor fixes on a case by case basis when we notice them

2. What I did to test those areas of impact (or what existing automated tests I relied on)
    - manual testing

3. What automated tests I added (or what prevented me from doing so)
    - n/a

PR submission checklist:
